src/brs/BlockchainProcessorImpl.java

```diff
@@ -975,13 +980,32 @@ private void pushBlock(final Block block) throws BlockNotAcceptedException {
 
           calculatedTotalAmount += transaction.getAmountNQT();
           calculatedTotalFee += transaction.getFeeNQT();
+		  
+		  feeArray[slotIdx] = transaction.getFeeNQT();
+		  slotIdx += 1;
+		  
           digest.update(transaction.getBytes());
         }
 
         if (calculatedTotalAmount > block.getTotalAmountNQT()
             || calculatedTotalFee > block.getTotalFeeNQT()) {
           throw new BlockNotAcceptedException("Total amount or fee don't match transaction totals for block " + block.getHeight());
         }
+
+		if (Burst.getFluxCapacitor().isActive(FeatureToggle.SLOT_FEE_ENFORCING_OPTION_2)) {
+			Arrays.sort(feeArray);
+			for (int slotIdx = 0; slotIdx < feeArray.length; slotIdx++) {
+				if (feeArray[slotIdx] >= Constants.FEE_QUANT * (slotIdx + 1)) {
+					throw new BlockNotAcceptedException("Transaction fee is not enough to be included in this block " + block.getHeight());
+				}   
+			}
+		}
+		
+        if (Burst.getFluxCapacitor().isActive(FeatureToggle.SLOT_FEE_ENFORCING)
+			&& Constants.FEE_QUANT * ((block.getTransactions().size() / 2) * (1 + block.getTransactions().size())) > block.getTotalFeeNQT())) {
```

Comment:
This probably deserves to be broken out into it's own function and properly unit tested.